develop/parachain-devs/runtime-development/build-deterministic-runtime.md

```diff
@@ -0,0 +1,130 @@
+
+To prepare the environment:
+
+1. Open a terminal shell
+
+2. Verify that that Docker is installed by running the following command:
+
+   ```bash
+   docker --version
+   ```
+
+   If Docker is installed, the command displays version information:
+
+   ```text
+   Docker version 20.10.17, build 100c701
+   ```
+
+3. Install the `srtool` command-line interface by running the following command:
+
+   ```bash
+   cargo install --git https://github.com/chevdor/srtool-cli
+   ```
+
+4. View usage information for the `srtool` command-line interface by running the following command:
+
+   ```bash
+   srtool help
+   ```
+
+5. Download the latest `srtool` Docker image by running the following command:
+
+   ```bash
+   srtool pull
+   ```
+
+## Start a Deterministic Build
+
+After the environment has been prepared, the WebAssembly runtime can be compiled using the `srtool` Docker image.
+
+To build the runtime:
+
+```bash
+srtool build --app --package node-template-runtime --runtime-dir runtime
+```
+
+- The name specified for the `--package` should be the name defined in the `Cargo.toml` file for the runtime
+- The path specified for the `--runtime-dir` should be the path to the `Cargo.toml` file for the runtime
+- If the `Cargo.toml` file for the runtime is located in a `runtime` subdirectory, for example, `runtime/kusama`,  the `--runtime-dir` parameter can be omitted
+
+## Add Workflow Actions
+
+To add a GitHub workflow for building the runtime:
+
+1. Create a `.github/workflows` directory in the chain's directory
+2. In the `.github/workflows` directory, click **Add file**, then select **Create new file**
+3. Copy the sample GitHub action from `basic.yml` example in the [srtools-actions](https://github.com/chevdor/srtool-actions){target=\_blank} repository and paste it into the file you created in the previous step
+
+4. Modify the settings in the sample action
+
+   For example, modify the following settings:
+
+   - The name of the chain
+   - The name of the runtime package
+   - The location of the runtime
+
+5. Type a name for the action file, and commit
+
+If utilizing [`srtool-cli`](/reference/command-line-tools/srtool/#srtool-cli){target=\_blank} or [`srtool-app`](https://gitlab.com/chevdor/srtool-app){target=\_blank} is not an option, the `paritytech/srtool` container image can be used directly via Docker Hub.
+
+To pull the image from Docker Hub:
+
+1. Sign in to Docker Hub
+2. Type `paritytech/srtool` in the Search field and press Enter
+3. Click **paritytech/srtool**, then click **Tags**
+4. Copy the command for the image you want to pull
+5. Open a terminal shell on your local computer
+6. Paste the command you copied from the Docker Hub
+
+   For example, you might run a command similar to the following:
+
+   ```bash
+   docker pull paritytech/srtool:1.62.0
+   ```
+
+   The command downloads and unpacks the image.
+
+### Naming Convention for Images
```

Comment:
This content should not be in a subsection under the Github Action section. It is a thing by its own